web/src/components/CountryDistribution/useCountryQuery.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect, useMemo } from 'react'
+import { useRouter } from 'next/router'
+import { Region, ClusterState, getPerCountryData, CountryDistribution } from 'src/io/getPerCountryData'
+import { Places } from 'src/io/getPlaces'
+
+import { getRegionBySelectedRegionQs, getCurriedClustersStateBySelectedClusters, ParsedUrlQuery } from './utils'
+
+
+export const convertUrlQueryToSelection = (queryString: ParsedUrlQuery): string[] => {
+  if (!queryString) {
+    return []
+  }
+  if (typeof queryString === 'string') {
+    return [queryString]
+  }
+  if (Array.isArray(queryString)) {
+    return queryString
+  }
+  return []
+}
+
+/**
+ * A react hook that take in router query params as source of truth
+ * and compute the React component states for UI display
+ */
+export const useRouterQuery = (): {
```

Comment:
main part of the PR. This hook abstract the previous logic, and also add logic to take in router's query params as source of truth, and use it to compute `region` (previously named `currentRegion`), `places`, `currentClusters`